### ISSUE TITLE
More control over document validator rules

### ISSUE THREAD

**Issue by adri** [webonyx/graphql-php#134]
Thanks a lot for this awesome library. 
I was wondering if there is a good way to have more control over what rules are used for validation. In my usecase I'm using this library with overblog/GraphqlBundle. 

When doing some profiling I noticed that some time is spent in validating the query. This is also ok. But when I use whitelisted queries that have been checked before, could I skip validating all rules? 

<img width="914" alt="screen shot 2017-06-21 at 17 37 06" src="https://user-images.githubusercontent.com/133832/27393168-d3aef144-56a8-11e7-8216-3d80621f4ac5.png">

Maybe this is also just a measuring mistake. I'll update this issue if I find out more.
Thanks again!

**Comment by vladar** [comment]
Yes, you can skip validation. `GraphQL::execute` is just a conveniece facade over parsing, validation and execution steps. You can unwrap it and create your own facade with custom validation logic.

```php
try {
    if ($requestString instanceof DocumentNode) {
        $documentNode = $requestString;
    } else {
        $source = new Source($requestString ?: '', 'GraphQL request');
        $documentNode = Parser::parse($source);
    }

    $queryComplexity = DocumentValidator::getRule('QueryComplexity');
    $queryComplexity->setRawVariableValues($variableValues);

    $validationErrors = DocumentValidator::validate($schema, $documentNode);

    if (!empty($validationErrors)) {
        return new ExecutionResult(null, $validationErrors);
    } else {
        return Executor::execute($schema, $documentNode, $rootValue, $contextValue, $variableValues, $operationName);
    }
} catch (Error $e) {
    return new ExecutionResult(null, [$e]);
} 
```

See the `DocumentValidator::validate()` call. It has 3rd argument which accepts an array of validation rules to be executed (or you can just skip `validate` step entirely for your whitelisted queries).

**Comment by adri** [comment]
Sorry for the late answer. Thanks a lot! That's awesome.

**Comment by adri** [comment]
Ops you wanted to keep the issue open :D sorry

**Comment by vladar** [comment]
New signature of `GraphQL::execute` (and `GraphQL::executeAndReturnResult`):

```php
    function execute(
        Schema $schema,
        $source,
        $rootValue = null,
        $contextValue = null,
        $variableValues = null,
        $operationName = null,
        callable $fieldResolver = null,
        array $validationRules = null,
    )
```

Setting `$validationRules` to empty array will let you skip the validation. Will be published in v0.10.0.

**Comment by adri** [comment]
Awesome! Thanks a lot @vladar :)
